shell/agents/Microsoft.Azure.Agent/Telemetry.cs

```diff
@@ -17,10 +17,10 @@ internal static void Initialize()
     {
         InstallationId = null;
 
-        string azureConfigDir = Environment.GetEnvironmentVariable("AZURE_CONFIG_DIR")
-            ?? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".azure");
-        string azCLIProfilePath = Path.Combine(azureConfigDir, "azureProfile.json");
-        string azPSHProfilePath = Path.Combine(azureConfigDir, "AzureRmContextSettings.json");
+        string azureConfigDir = Environment.GetEnvironmentVariable("AZURE_CONFIG_DIR");
+        string userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+        string azCLIProfilePath = azureConfigDir ?? Path.Combine(userProfile, ".azure", "azureProfile.json");
+        string azPSHProfilePath = azureConfigDir ?? Path.Combine(userProfile, ".Azure", "AzureRmContextSettings.json");
```

Comment:
`azureConfigDir` is a folder path, so when `AZURE_CONFIG_DIR` is defined in environment variable, `azCLIProfilePath` and `azPSHProfilePath` will be set to the folder path in your code.